File: app/src/main/java/com/example/fooddelivery/MainActivity.java
package com.example.fooddelivery;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btn = (Button)findViewById(R.id.btStart);
        btn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        TextView tvPhoneNumber = (TextView)findViewById(R.id.tvPhoneNumber);
        Intent myIntent = new Intent(getApplicationContext(), BasketOrdersActivity.class);
        myIntent.putExtra("phoneNumber", tvPhoneNumber.getText().toString());
        startActivity(myIntent);
    }
}

File: app/src/main/java/com/example/fooddelivery/MainActivity.kt
package com.example.fooddelivery

import android.os.Bundle
import androidx.fragment.app.FragmentActivity
import com.example.fooddelivery.fragments.AuthFragment
import com.example.fooddelivery.fragments.MenuDetailDialogFragment
import com.example.fooddelivery.fragments.OrdersFragment
import com.example.fooddelivery.fragments.PagerFragment

class MainActivity : FragmentActivity(), Navigation {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupAuthFragment()
    }

    override fun setupAuthFragment() {
        supportFragmentManager
            .beginTransaction()
            .add(R.id.rootScreen, AuthFragment())
            .commit()
    }

    override fun setupOrdersFragment(phoneNumber: String) {
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.rootScreen, OrdersFragment.getNewInstance(phoneNumber))
            .addToBackStack(OrdersFragment.NAME_TAG)
            .commit()
    }

    override fun openMenuDialog() {
        MenuDetailDialogFragment()
            .show(supportFragmentManager, MenuDetailDialogFragment.NAME_TAG)
    }

    override fun openViewPager() {
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.rootScreen, PagerFragment())
            .addToBackStack(OrdersFragment.NAME_TAG)
            .commit()
    }
}